.gitignore

```diff
@@ -1,5 +1,6 @@
 # Hugo
-public/
+public/*
+!public/.gitkeep
```

Comment:
This will allow the public directory to be in the repo but ignore all other files and subdirectories. 

The `!public/.gitkeep` is to allow the gitkeep file to be there and in turn makes git keep the empty directory